pyensign/connection.py

```diff
@@ -65,68 +66,146 @@ def __init__(self, connection):
             The connection to the Ensign server
         """
 
-        self.stub = ensign_pb2_grpc.EnsignStub(connection.channel)
+        self.channel = connection.channel
+        self.stub = ensign_pb2_grpc.EnsignStub(self.channel)
+        self.publish_streams = {}
+        self.subscribe_streams = {}
+        self.pool = WorkerPool(max_workers=10, max_queue_size=100)
 
     @catch_rpc_error
-    async def publish(self, topic_id, events, client_id=""):
-        def next():
-            # First message must be an OpenStream request
-            # TODO: Should we send topics here?
-            yield ensign_pb2.PublisherRequest(
-                open_stream=ensign_pb2.OpenStream(client_id=client_id)
+    async def publish(
+        self, topic_id, events, ack_callback=None, nack_callback=None, client_id=""
+    ):
+        # Create a hash of the topic ID
+        topic_hash = str(topic_id)
```

Comment:
This might have to be a lookup in the topic maps that should get stored on this class.